gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/javaversions/BaselineJavaVersion.java

```diff
@@ -267,6 +265,16 @@ public final void checkJavaVersions() {
         }
     }
 
+    /**
+     * <a href="https://github.com/JetBrains/intellij-community/pull/2135">Further details available here.</a>
+     */
+    private static void enablePreviewOnCompileTask(
+            Provider<ChosenJavaVersion> provider, CompileOptions compileOptions) {
+        if (provider.get().enablePreview()) {
```

Comment:
This is eager and not lazy, obviously. @iamdanfox would this be a problem?

Reply:
The downside of this 'eager' approach is:

If the first line of your build.gradle says `apply plugin: com.palantir.baseline-version` and then your second line says: `javaVersions { runtime = '19_PREVIEW' }`, then we will already have configured this `compileOptions` task and _not_ added the `--enable-preview` compiler arg.

Reply:
Yeah this should be lazy. You can use `getCompilerArgumentProviders().add` to give an argument provider that can return a list of size 0 or 1 to be lazy.

Reply:
@CRogers the compilerArgumentProviders approach is what's already on develop, but sadly when you use IntelliJ's native integration, they literally check `javaCompileTask.options.compilerArgs.contains("--enable-preview")` which does _not_ invoke my beloved command line argument provider. 

James has a fix up here: https://github.com/JetBrains/intellij-community/pull/2135

Reply:
Yeah I probably should have read that first 🤦🏻

Reply:
ok folks. I've updated this PR to actually use both methods (lazy and eager). Dan and I think that this will _most likely_ solve the problem in all practical cases because in any case where code is in subprojects, the javaVersions closure will have been set on the root and so evaluated before the java library declaration, avoiding the problem.

However, just in case laziness is an issue (e.g. you have only a root project) this will maintain the fact that --enable-preview is set, with the only downside being it's most likely set twice. This should be a no-op in all cases, should guarantee no semantic changes from present _except_ avoiding the IntelliJ bug.